auth/src/main/java/org/aerogear/mobile/security/SecurityCheckExecutor.java

```diff
@@ -5,112 +5,158 @@
 import android.support.annotation.NonNull;
 import android.support.annotation.Nullable;
 
+import org.aerogear.mobile.core.executor.AppExecutors;
 import org.aerogear.mobile.core.metrics.MetricsService;
 
 import java.util.Collection;
 import java.util.HashSet;
 
+import static org.aerogear.mobile.core.utils.SanityCheck.nonNull;
+
 /**
  * Entry point for the SecurityCheckExecutor.
  * This class provides the builders.
  */
 public class SecurityCheckExecutor {
-    private SecurityCheckExecutor() {
-
-    }
+    private SecurityCheckExecutor() {}
 
     /**
      * Entry point for SecurityCheckExecutor builders.
      */
     public static class Builder {
 
         /**
-         * Base class for SecurityCheckExecutor builders
+         * Base class for SecurityCheckExecutor builders.
+         *
          * @param <T> The type of this builder
          * @param <K> The type of the built object
          */
         static abstract class AbstractBuilder<T, K> {
+
             private final Context ctx;
             private final Collection<SecurityCheck> checks = new HashSet<>();
             private MetricsService metricsService;
 
+            /**
+             * Creates AbstractBuilder object.
+             *
+             * @param ctx {@link Context} to be used by the security checks
+             * @throws IllegalArgumentException if ctx is null
+             */
             public AbstractBuilder(@NonNull final Context ctx) {
-                this.ctx = ctx;
+                this.ctx = nonNull(ctx, "context");
```

Comment:
This should be a defensive getApplicationContext so we don't leak reference to Activities.